tests/js/BorrowerOperationsTest.js

```diff
@@ -6995,6 +6996,35 @@ contract("BorrowerOperations", async accounts => {
 
         assert.isTrue(newTCR.eq(expectedTCR));
       });
+
+      it("openTrove(): open a Trove, then adjust it in the same block should revert", async () => {
+        await openTrove({ ICR: toBN(dec(2, 18)), extraParams: { from: alice } });
+        await priceFeed.setPrice("105000000000000000000");
+
+        assert.isTrue(await th.checkRecoveryMode(contracts));
+
+        const extraZUSDAmount = toBN(dec(1000, 18));
+        const MIN_DEBT = (
+          await th.getNetBorrowingAmount(contracts, await contracts.borrowerOperations.MIN_NET_DEBT())
+        ).add(th.toBN(1)); // add 1 to avoid rounding issues
+        const zusdAmount = MIN_DEBT.add(extraZUSDAmount);
+
+        const price = await contracts.priceFeedTestnet.getPrice();
+        const totalDebt = await th.getOpenTroveTotalDebt(contracts, zusdAmount);
+        const ICR = toBN(dec(2, 18));
+        const borrowerOperationsCrossReentrancy = await BorrowerOperationsCrossReentrancy.new(contracts.borrowerOperations.address);
+
+        await assertRevert(
+          borrowerOperationsCrossReentrancy.testCrossReentrancy(
```

Comment:
please check the test and implementation:
1) you already put the protocol into maintenance mode, but the function does it again in the borrowerOperationsCrossReentrancy contract   
2) in the borrowerOperationsCrossReentrancy you just add collateral - it should not revert because only closing/decreasing debt in the same block as opening/increasing should revert
3) consider removing the mock contract because you can simulate processing in the same block via modifying the hardhat [mining mode](https://hardhat.org/hardhat-network/docs/explanation/mining-modes#mining-multiple-transactions-in-one-block)

Reply:
1. Actually we don't  need the instruction to put the state to the recovery mode anymore in borrowerOperationsCrossReentract contract. [Removed](https://github.com/DistributedCollective/zero-contracts/pull/7/commits/bd697a6e2e22bd6012b02197c539832b763f6153) it 

2. Isn't The Adding the collateral will adjust the debt (decreasing debt) and it is done in the same block with the opening with the [trove opening](https://github.com/DistributedCollective/zero-contracts/blob/9c5e883a447ff11758cdee1cae220e76b3fabca7/contracts/TestContracts/BorrowerOperationsCrossReentrancy.sol#L28)

3. I am aware about this, but still, i'd prefer to use mock, since it's more explicit

Reply:
2. adding collateral does not affect debt - the debt is the ZUSD amount borrowed against collateral.    
 3. i see what you mean but the mock contract just adds collateral in the same tx which should not revert.

Reply:
you are right, just updated the unit & contract test in this updated in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/7/commits/87907c748eda0da187bd5f5ed5f48f5d4c511b4e)